### ISSUE TITLE
Handling CTE at the start of query in DML

### ISSUE THREAD

**Issue by simpl1g** [reata/sqllineage#328]
Given that query, I expect that `model` should be in `cte` section. 
Is it some kind of reserved word?

```sql
INSERT INTO db.table1 ("date")
WITH raw AS (
  SELECT
    date
  FROM db.table2
), model AS (
  SELECT
    raw.date
  FROM raw
  LEFT JOIN db.table3 USING (date)
)
SELECT * FROM model

table read: [Table: <default>.model, Table: db.table2, Table: db.table3]
table write: [Table: db.table1]
table cte: []
```

When there is only one CTE it works correctly
```sql
INSERT INTO db.table1 ("date")
WITH model AS (
  SELECT
    db.date
  FROM db.table2
)
SELECT * FROM model

table read: [Table: db.table2]
table write: [Table: db.table1]
table cte: [SubQuery: model]
```

If I wrap `raw` in backticks it also works

```sql
INSERT INTO db.table1 ("date")
WITH `raw` AS (
  SELECT
    date
  FROM db.table2
), model AS (
  SELECT
    raw.date
  FROM raw
  LEFT JOIN db.table3
    USING (date)
)
SELECT * FROM model

table read: [Table: db.table2, Table: db.table3]
table write: [Table: db.table1]
table cte: [SubQuery: `raw`, SubQuery: model]
```

versions that I use:
```
sqllineage               1.3.7
sqlparse                 0.4.1
```

**Comment by reata** [comment]
This issue is written in a way nothing short of exemplary. Thank you for making debugging so much easier.

To the problem, you're right. RAW being keyword as defined by sqlparse changes the behavior of the parsing result. This is a known problem causing so much other issues like #183 #93  that we decide to replace the parser. 

We have a huge PR working in progress right now to introduce sqlfluff as parser. Hopefully we shall be able to release it in next few months. Then all the keywords related problem should be solved.

In short term, as I understand, RAW is a keyword only in Oracle. So if you're not using RAW with Oracle, you can disable this keyword with following code before calling sqllineage.
```
from sqlparse.keywords import KEYWORDS
del KEYWORDS["RAW"]
```

**Comment by simpl1g** [comment]
Thank you for quick workaround! We are using ClickHouse, so it works fine. 

Cudos for `sqlfluff`. We already use it as linter.

BTW, maybe you know how to extend lineage so it can support table functions as sources?
For example in ClickHouse we often do something like this to read from PG:
```sql
SELECT column_from_pg
FROM postgresql(connection_id, table = 'user_profiles')
```

**Comment by reata** [comment]
With current sqlparse implementation, you might want to check https://github.com/reata/sqllineage/blob/master/sqllineage/core/handlers/source.py#L82 to change the code when token after FROM keyword is a function.

For the moment, feel free to monkey patch or change the code and then build you own distribution. After the sqlfluff PR is merged into master, a PR with new implementation is more than welcome.

**Comment by simpl1g** [comment]
Great, thank you for your help!

**Comment by reata** [comment]
v1.4.x release already fixed the keyword related issue. However, it introduces regression when handling CTE, which we need to fix.

**Comment by reata** [comment]
After we release v1.4.3, you shall be able to use ansi dialect to get the correct lineage:

```sql
INSERT INTO db.table1 ("date")
WITH raw AS (
  SELECT
    date
  FROM db.table2
), model AS (
  SELECT
    raw.date
  FROM raw
  LEFT JOIN db.table3 USING (date)
)
SELECT * FROM model
```

```shell
$ sqllineage -f test.sql --dialect=ansi -v
Statement #1: INSERT INTO db.table1 ("date")WITH raw AS (  SELEC...
    table read: [Table: db.table2, Table: db.table3]
    table write: [Table: db.table1]
    table cte: [SubQuery: model, SubQuery: raw]
    table drop: []
    table rename: []
==========
Summary:
Statements(#): 1
Source Tables:
    db.table2
    db.table3
Target Tables:
    db.table1
```
